app/scripts/lib/router.js

```diff
@@ -86,37 +80,17 @@ function (
 
   function showView(View, options) {
     return function () {
-      // If the current view is an instance of View, that means we're
-      // navigating from a subview of the current view
-      if (this.currentView instanceof View) {
-        this.trigger(this.NAVIGATE_FROM_SUBVIEW, options);
-        this.setDocumentTitle(this.currentView.titleFromView());
-      } else {
-        this.createAndShowView(View, options);
-      }
+      return this.showView(View, options);
     };
   }
 
-  // Show a sub-view, creating and initializing the SuperView if needed.
-  function showSubView(SuperView, options) {
+  function showSubView(SubView, SuperView, options) {
```

Comment:
@philbooth - What do you think if I rename the constructors to `SubView` and `ParentView`? I kinda would prefer `ChildView` and `ParentView` but that'll require a lot of function name updates as well and could be done later.